feat: write about using Comparator and trusting our statements (#595)
fpvandoorn/carleson: paper/main.tex

The proof of this last result is the heart of the matter. The pieces are categorized in one of three types, for each of which a separate analysis is required to establish a bound.
These three branches of the proof are largely independent and handled in separate sections 5, 6, and 7 of the blueprint \cite{CarlesonBlueprint}.

> \subsection{Verification of results}

Reviewer: Should this be in Design decisions? Project organization? 
Let's discuss in-person.

Author: As discussed in person: having this in this section is fine, and there should be a new section 2.3 before it.
```suggestion
\subsection{Verification the definitions and statements}
```

> The Lean kernel checks that all proofs ar eocrrect. It cannot ascertain that all definitions and theorems match their definition in the blueprint; this requires manual effort.

Reviewer: ```suggestion
The Lean kernel checks that all proofs are correct. It cannot ascertain that all definitions and theorems match their definition in the blueprint; this requires manual effort.
```

> One strategy (going back to at least the Liquid Tensor experiment \cite{LTEDefinitions}) is to prove interesting theorems about these definitions (such as Carleson's theorem).

Reviewer: ```suggestion
One strategy (going back to at least the Liquid Tensor Experiment \cite{LTEDefinitions}) is to prove interesting theorems about these definitions (such as Carleson's theorem).
```
(or all lowercase)

Reviewer: Also, specify what you mean by Carleson's theorem, e.g. "(such as Carleson's classical theorem, \Cref{classical-carleson})."

Author: ```suggestion
One strategy (going back to at least the Liquid Tensor Experiment \cite{LTEDefinitions}) is to prove interesting theorems about these definitions (such as Carleson's classical theorem, \Cref{classical-carleson}).
```

> Another aspects is making the relevant definitions easy to audit: to this end, we have collected all definitions and set-up necessary for stating the main theorems into a single file. Auditing these 160 lines of Lean code (on top of \Mathlib) ensures the theorems say what they claim --- two orders of magnitude small than the entire project.

Reviewer: ```suggestion
Another aspects is making the relevant definitions easy to audit: to this end, we have collected all definitions and set-up necessary for stating the main theorems into a single file. Auditing these 160 lines of Lean code (on top of \mathlib) ensures the theorems say what they claim --- two orders of magnitude smaller than the entire project.
```

% 137 lines of Lean code (without comments) in BasicDefinitions, 26 in Challenge

> In addition, about a quarter of these lines are basic definitions that will be added to Mathlib soon --- reducing the code to be trusted further.

Reviewer: > reducing the code to be trusted further.

Perhaps better phrasing: "so that these definitions get a careful review from the Mathlib reviewers" (it's still trusted code(?))

Author: ```suggestion
In addition, about a quarter of these lines are basic definitions that will be added to \mathlib soon --- which implies getting careful review (and reducing the need to trust code beyond \mathlib further).
```

Author: What do you think about this phrasing?

Reviewer: sounds good to me

> The astute reader may argue that the \emph{proofs} of these results require many more prerequisites, complicating this analysis: how can we ensure the actually proven statements match their theorems? We offer two answers:

Reviewer: I don't particularly like this phrasing, but if you like it, feel free to keep it.

I think that most readers will know that this is what Lean does, so I would just write something like "Lean checks that the formalized proof is a valid proof of the given statement.

Author: Good points! I have reworded that paragraph entirely.
